backend/src/main/java/application/controller/TaskController.java

```
@@ -50,4 +51,11 @@ Task setTaskFinishedValue(@RequestBody boolean checked, @PathVariable Long id) {
     public void deleteTask(@PathVariable Long id) {
         taskService.deleteTask(id);
     }
+
+    @GetMapping("/task/{id}")
+    public Task getTask(@PathVariable Long id) {
+
+        return taskService.getTask(id)
+                .orElseThrow(() -> new TaskNotFoundException(id));
```

We can throw this exception in all controller methods in which we search for a task.